### ISSUE TITLE
eslint not analyzing props name when declared in brackets

### ISSUE THREAD

**Issue by hyphaene** [eslint/eslint#9835]
**Tell us about your environment**

* **ESLint Version:**
4.12.0
* **Node Version:**
8.4.0
* **npm Version:**
5.3.0
**What parser (default, Babel-ESLint, etc.) are you using?**
 Babel-ESLint
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js

{
    "parser": "babel-eslint",
    "env": {
        "jest/globals": true
    },
    "extends": ["airbnb"],
    "plugins": ["jest", "import"],
    "rules": {
        "indent": ["error", 4, { "SwitchCase": 1 }],
        "import/prefer-default-export": "error",
        "jsx-a11y/no-static-element-interactions": "off",
        "react/jsx-indent": ["error", 4],
        "react/jsx-indent-props": ["error", 4],
        "import/no-absolute-path": "off",
        "jsx-a11y/click-events-have-key-events": "off",
        "jsx-a11y/anchor-is-valid": "off",
        "no-plusplus": ["error", { "allowForLoopAfterthoughts": true }],
        "function-paren-newline": "off",
        "object-curly-newline": "off"
    },
    "globals": {
        "document": true,
        "window": true,
        "navigator": true,
        "localStorage": true,
        "fetch": true
    },
    "settings": {
        "import/resolver": {
            "webpack": {
                "config": "eslint-webpack.config.js"
            }
        }
    }
}


```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
import React, { Component } from 'react';
import PropTypes from 'prop-types';

const CLIENTS_LIST = 'clientsList';

// import './example.scss

export default class CreateClientModalComponent extends Component {
    getClientsList() {
        return this.props[CLIENTS_LIST];
    }
    render() {
        const { clientsList } = this.props;
        if (clientsList.length) return <div className="create-user" />;
        return null;
    }
}

CreateClientModalComponent.propTypes = {
    [CLIENTS_LIST]: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
};





```

```bash
directly in the VSCODE editor window ( on hover  on the underlined-errored keys )
```

**What did you expect to happen?**

I expected that eslint parse the value into brackets to understand that 
clientsList and [CLIENTS_LIST] but the notation in brackets is not properly handled, eslint only read the characters written and does not get the real value of the key.
**What actually happened? Please include the actual, raw output from ESLint.**

on hover =>
[eslint] 'CLIENTS_LIST' PropType is defined but prop is never used (react/no-unused-prop-types)
AND
[eslint] 'clientsList' is missing in props validation (react/prop-types)
 and in console =>
15:27  error  'clientsList' is missing in props validation               react/prop-types
  27:21  error  'CLIENTS_LIST' PropType is defined but prop is never used  react/no-unused-prop-types

**Comment by j-f1** [comment]
This looks like it’s an issue with [`eslint-plugin-react`](https://github.com/yannickcr/eslint-plugin-react), not with ESLint itself.

**Comment by hyphaene** [comment]
Ok thanks for the redirection, I'll check with them.

**Comment by platinumazure** [comment]
Closing as this is not an issue with ESLint core. Thanks!
